pkg/kubernetes/scripts/storage.js

```diff
@@ -0,0 +1,766 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function() {
+    "use strict";
+
+    var VOLUME_FACTORY_SUFFIX = "VolumeFields";
+
+    angular.module('kubernetes.storage', [
+        'ngRoute',
+        'ui.cockpit',
+        'kubeClient',
+        'kubernetes.listing',
+        'ui.cockpit',
+    ])
+
+    .config([
+        '$routeProvider',
+        function($routeProvider) {
+            $routeProvider
+                .when('/storage', {
+                    templateUrl: 'views/pv-listing.html',
+                    controller: 'StorageCtrl'
+                })
+
+                .when('/storage/:target', {
+                    controller: 'StorageCtrl',
+                    templateUrl: 'views/pv-page.html'
+                });
+        }
+    ])
+
+    /*
+     * The controller for the storage view.
+     */
+    .controller('StorageCtrl', [
+        '$scope',
+        'kubeLoader',
+        'kubeSelect',
+        'ListingState',
+        'filterService',
+        '$routeParams',
+        '$location',
+        'storageActions',
+        '$timeout',
+        function($scope, loader, select,  ListingState, filterService,
+                 $routeParams, $location, actions, $timeout) {
+            var target = $routeParams["target"] || "";
+            $scope.target = target;
+
+            var c = loader.listen(function() {
+                var timer;
+                $scope.pvs = select().kind("PersistentVolume");
+                if (target) {
+                    $scope.item = select().kind("PersistentVolume").name(target).one();
+                    if (!$scope.item && !timer) {
```

Comment:
I think we need to make some sort of "Not found" display rather than redirecting when something doesn't exist. What do you think? Perhaps as a follow up?